base_layer/core/src/base_node/states/block_sync.rs

```diff
@@ -189,6 +192,10 @@ async fn synchronize_blocks<B: BlockchainBackend + 'static>(
             info!(target: LOG_TARGET, "Synchronize missing blocks.");
             for height in sync_height..=network_tip_height {
                 request_and_add_block(db, comms, config, sync_peers, height).await?;
+                if height == network_tip_height {
+                    info!(target: LOG_TARGET, "Check if sync peer chain has been extended.");
+                    network_tip_height = request_network_tip_height(comms, config, sync_peers).await?;
```

Comment:
See clippy warning; move this outside the loop rather

Reply:
I changed it rather to a While loop so it repeats the tip update check until the local and network chain converge to the same height.